### PR TITLE
Webpack/Typescript/Angular Intro Task

### PR DESCRIPTION
**Pull request by ipelekhan**
- [x] Add initial components
- [x] Add user and course entities

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -3,10 +3,26 @@ import { NgModule } from '@angular/core';
 
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+import { HeaderComponent } from './header/header.component';
```
Comment (dragonflypl) [inline src/app/app.module.ts:6]: Nice separation of components. 

One remark, try from the beginning to establish a folder structure, don't put everything into one folder.

Also, try to use modules read about (basically everything about modules from the docs):

- https://angular.io/guide/module-types#shared-ngmodules
- https://angular.io/guide/module-types#widget
- https://angular.io/guide/feature-modules
- https://angular.io/guide/ngmodules

Using modules from the very beginning greatly simplifies ongoing development including writing tests, reusing the code etc...

File: src/app/breadcrumbs/breadcrumbs.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { BreadcrumbsComponent } from './breadcrumbs.component';
+
+describe('BreadcrumbsComponent', () => {
```
Comment (dragonflypl) [inline src/app/breadcrumbs/breadcrumbs.component.spec.ts:5]: For testing, in the future, please try to use:

https://testing-library.com/docs/angular-testing-library/intro

There's no better way to test angular components

**Review by dragonflypl — APPROVED** [review]
Please add classes to satisfy the homework requirements...

File: src/app/breadcrumbs/breadcrumbs.component.spec.ts (hunk 2 above)
Comment (ipelekhan) [reply src/app/breadcrumbs/breadcrumbs.component.spec.ts:5]: Will add with the next PR.

**Pull request merged**